src/reducetensor.cpp

```diff
@@ -568,7 +568,7 @@ std::size_t ReduceTensorDescriptor::GetWorkspaceSize(const Handle& handle,
     std::size_t wsSizeInBytes =
         !need_indices ? workspace_size * detail::GetDataTypeSize(inDesc.GetType())
                       : workspace_size * (detail::GetDataTypeSize(inDesc.GetType()) + sizeof(int)) +
-                            64 + sizeof(int);
+                            64 + sizeof(int) + sizeof(int);
```

Comment:
Am I right and you wanted to cover any misalignment for the buffer with additional `sizeof(int)`?
I guess it's better to use a single global constant for the buffer and for the further alignment routine.

Reply:
The workspace is divided into 4 buffers inside the kernel.

Reply:
It looks like each of them should be aligned properly.
Actually, I'm going to do some sort of `workspace` container which will hold multiple properly aligned raw buffers from the workspace.
But I'm not sure about timeline and what it will be adopted by all the algorithms.